controllers/hybrisapp_controller.go

```diff
@@ -344,9 +345,11 @@ func (r *HybrisAppReconciler) createBuildConfigForHybrisApp(hybrisApp *hybrisv1a
 				Strategy: buildv1.BuildStrategy{
 					Type: buildv1.SourceBuildStrategyType,
 					SourceStrategy: &buildv1.SourceBuildStrategy{
-						From: corev1.ObjectReference{
-							Kind: "ImageStreamTag",
-							Name: strings.Join([]string{hybrisApp.Spec.BaseImageName, hybrisApp.Spec.BaseImageTag}, ":"),
+						From: corev1.ObjectReference{Kind: "ImageStreamTag", Name: strings.Join([]string{hybrisApp.Spec.BaseImageName, hybrisApp.Spec.BaseImageTag}, ":")},
+						Env: []corev1.EnvVar{
```

Comment:
You only covered the scenario when a new buildconfig is created. 
In the function `ensureBuildConfig`, you should also check if user updates the settings of a deployed app. If settings are updated, you should update the buildconfig and make sure a new build starts to rebuild the app image, and also the deploymentconfig redeploys the app with the new image. 
This is the scenario to test for changing settings.
By the way, the app image will be rebuilt every time the user changes some local properties, it is an issue of the approach.

Reply:
This design is nothing new from before. I do not want to change the existing behavior. We can handle an architectural change as this becomes a product

Reply:
When you are free get on a call with me. Show me exactly what you are trying to refer to here? I do not see any changes in the update of the BC design. The trigger for them is as before

Reply:
It is here in `ensureBuildConfig`: https://github.com/faizalak/sap-commerce-operator/blob/0c23f99509f1860540355627c8ddc7cae713836a/controllers/hybrisapp_controller.go#L291

Reply:
I understand your point now. I think it is alright to proceed since I am not expecting the HybriApp to be updated later to the initial creation from the operator. These new properties need not trigger an update. Just creation should suffice for now.
I will hold this change for a future date if the customer comes back asking for anything further. For now, I need the priority to have the approach tested to see if it fits the customer